scripts/qesap/test/unit/test_qesap_lib_cmd_ansible.py

```diff
@@ -0,0 +1,46 @@
+from unittest import mock
+import os
+import logging
+import yaml
+
+from lib.cmds import cmd_ansible
+
+log = logging.getLogger(__name__)
+
+
+@mock.patch("lib.process_manager.subprocess_run")
+def test_cmd_ansible(subprocess_run, tmpdir):
+    '''
+    This test coverage overlap with tests from
+    scripts/qesap/test/unit/test_qesap_ansible.py
+    
+    '''
+
+    # Set env and input
+    conf_yaml = '''---
+apiver: 3
+provider: "lolo"
+terraform:
+  variables:
+    sandwiches:
+      - club
+      - cheese
+'''
+    data = yaml.load(conf_yaml, Loader=yaml.FullLoader)
+    provider_folder = tmpdir / 'ansible'
+    os.makedirs(provider_folder)
+    subprocess_run.return_value = (0, ['This is the ansible output', 'Two lines of that'])
+
+    # Set expectation
+    calls = []
+    ansible_cmd = [
+        'ansible',
+        f"-chdir={provider_folder}"]
+    # Just test one of them
+    ansible_cmd.append('-i')
+    ansible_cmd.append('-vvvv')
+    calls.append(mock.call(ansible_cmd))
+
+    ret = cmd_ansible(data, tmpdir, False, True)
+
+    assert ret == 1
```

Comment:
yes it is expected because return the file "ansible.all.log.txt" the generic one

Reply:
> yes it is expected because return the file "ansible.all.log.txt" the generic one
Could you please elaborate? `cmd_ansible` return the exit code of the executed commands. As this test code has

```
subprocess_run.return_value = (0, ['This is the ansible output', 'Two lines of that'])
```

it should return that 0

Reply:
in this new UT there's

```
@mock.patch("lib.process_manager.subprocess_run")
```

It means, "Before to call cmd_ansible , replace the true subprocess_run with a fake version of it."


and then
```
subprocess_run.return_value = (0, ['This is the ansible output', 'Two lines of that'])
```

that means, "the fake version of it, when called, just return (0, ['This is the ansible output', 'Two lines of that'])  that means "exitcode 0" from subprocess_run and two lines of stdout"

but if `cmd_ansible` calls `subprocess_run` and it return 0, also `cmd_ansible` should return 0, and not 1

Reply:
Refactored

Reply:
my point was that you should change test from
assert ret == 1
to
assert ret == 0
and if it fails debug why ret is not 0, as it should

Reply:
updated